miasm2/expression/expression_dissector.py

```diff
@@ -0,0 +1,168 @@
+import miasm2.expression.expression as m2_expr
+
+
+class ExprDissector:
+    """
+    Dissects expressions into subexpressions
+    """
+
+    def __init__(self, registers=None, irdst=None):
+        """
+        Initialises the architecture variables
+        :param registers: registers of an architecture
+        :param irdst: IRA instruction pointer of an architecture
+        """
+        # set architecture variables
+        if registers and irdst:
+            self.arch_vars = set(registers + [irdst])
+
+    def _iterator(self, e):
+        """
+        Parses the arguments of an expression
+        :param e: expression
+        :return: iterator of a list of expressions
+        """
+        if isinstance(e, m2_expr.ExprOp):
+            iterator = iter(e.args)
+        elif isinstance(e, m2_expr.ExprSlice):
+            iterator = iter([e.arg])
+        elif isinstance(e, m2_expr.ExprCond):
+            iterator = iter((e.cond, e.src1, e.src2))
+        elif isinstance(e, m2_expr.ExprCompose):
+            iterator = iter((t[0] for t in e.args))
+        elif isinstance(e, m2_expr.ExprMem):
+            iterator = iter([e.arg])
+        elif isinstance(e, m2_expr.ExprAff):
+            iterator = iter([e.dst, e.src])
+        elif isinstance(e, m2_expr.ExprInt):
+            iterator = iter([])
+        elif isinstance(e, m2_expr.ExprId):
+            iterator = iter([])
+        else:
+            raise NotImplementedError()
+
+        return iterator
+
+    def _check_itetator(self, iterator, e):
```

Comment:
As this method is used only once, you can in-line it in `_dissect`.
Is `iteTator` a typo?

According to [Python doc](https://docs.python.org/2/library/exceptions.html#exceptions.NotImplementedError), `NotImplementedError` is reserved to indicate abstract methods in abstract classes (a bit misleading). You can raise, for instance, a `RuntimeError` instead.